libs/platform/ebpf_ring_buffer.h

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "ebpf_platform.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    typedef struct _ebpf_ring_buffer_record
+    {
+        size_t length;
+        // Set to either 0 or the epoch this was reserved in.
+        // If 0, then then this is a valid record.
+        // If MAXSIZE_T, then this is a discarded record.
+        // Else, this is a reserved record that hasn't been commit or discarded yet.
```

Comment:
```suggestion
        // Else, this is a reserved record that hasn't been submitted or discarded yet.
```
(or "committed", but the helper function uses the word "submit" rather than commit)